core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpServiceBuilder.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.dynamic;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ForkJoinPool;
+
+import com.google.common.collect.Sets;
+
+import com.linecorp.armeria.common.http.HttpMethod;
+
+/**
+ * Builds a new {@link DynamicHttpService}.
+ */
+public final class DynamicHttpServiceBuilder {
+
+    private Executor executor = ForkJoinPool.commonPool();
+    private final Map<Class<?>, ResponseConverter> converters = new HashMap<>();
+    private final List<DynamicHttpFunctionEntry> entries = new ArrayList<>();
+
+    /**
+     * Sets default {@link Executor} for asynchronous calls. If not set, it is performed using the
+     * {@link ForkJoinPool#commonPool()}.
+     */
+    public DynamicHttpServiceBuilder executor(Executor executor) {
+        this.executor = executor;
+        return this;
+    }
+
+    /**
+     * Adds a mapping from {@link Class} to {@link ResponseConverter}.
+     */
+    public DynamicHttpServiceBuilder convert(Class<?> type, ResponseConverter converter) {
```

Comment:
> Yes, `converter` or `addConverter` sound OK to me. A builder is often meant to keep add something until `build()`, so I think it's fine to omit `set` or `add` prefix.

Understood. I changed the method name into `addConverter`, which corresponds with `addMapping(s)` in this update.

> Could you explain why it has to remain to be a functional interface?

The reason I decided to make it a functional interface is that It would be better if it can be defined using lambda. That's all.